docs/hazmat/primitives/key-derivation-functions.rst

```diff
@@ -600,6 +600,131 @@ Different KDFs are suitable for different tasks such as:
         raises an exception if they do not match.
 
 
+.. currentmodule:: cryptography.hazmat.primitives.kdf.kbkdf
+
+.. class:: KBKDFHMAC(algorithm, mode, length, rlen, llen, location,\
+           label, context, fixed, backend)
+
+    .. versionadded:: 1.3
+
+    KBKDF (Key Based Derivation Function) is defined by NIST SP 800-108
+    in the `NIST-SP-800-108`_ document, to be used to derive additional
+    keys from a key that has been established through an automated
+    key-establishment scheme.
+
+    .. warning::
+
+        KBKDFHMAC should not be used for password storage.
+
+    .. doctest::
+
+        >>> import os
+        >>> from cryptography.hazmat.primitives import hashes
+        >>> from cryptography.hazmat.primitives.kdf.kbkdf import KBKDFHMAC
+        >>> from cryptography.hazmat.backends import default_backend
+        >>> backend = default_backend()
+        >>> label = b"KBKDF HMAC Label"
+        >>> context = b"KBKDF HMAC Context"
+        >>> kdf = KBKDFHMAC(
+        ...     algorithm=hashes.SHA256(),
+        ...     mode=KBKDFHMAC.COUNTER_MODE,
+        ...     length=256,
+        ...     rlen=4,
+        ...     llen=4,
+        ...     location=KBKDFHMAC.LOCATION_BEFORE_FIXED,
+        ...     label=label,
+        ...     context=context,
+        ...     fixed=None,
+        ...     backend=backend
+        ... )
+        >>> key = kdf.derive(b"input key")
+        >>> kdf = KBKDFHMAC(
+        ...     algorithm=hashes.SHA256(),
+        ...     mode=KBKDFHMAC.COUNTER_MODE,
+        ...     length=256,
+        ...     rlen=4,
+        ...     llen=4,
+        ...     location=KBKDFHMAC.LOCATION_BEFORE_FIXED,
+        ...     label=label,
+        ...     context=context,
+        ...     fixed=None,
+        ...     backend=backend
+        ... )
+        >>> kdf.verify(b"input key", key)
+
+    :param algorithm: An instance of a
+        :class:`~cryptography.hazmat.primitives.hashes.HashAlgorithm`
+        provider
+
+    :param str mode: The desired mode of PRF. If ``None`` is explicitly
```

Comment:
Would you mind changing this to an enum rather than a string (`KBKDFHMACMode.COUNTER` or something like that)? We generally try to avoid string types.